File: problem/Make_Fancy_String/makeFancyString.java
class Solution {
    public String makeFancyString(String s) {
        StringBuilder answer = new StringBuilder();

        for (int i = 0; i < s.length(); i++) {
            char c = s.charAt(i);
            int len = answer.length();

            if (len <= 1 || !(answer.charAt(len - 1) == c && answer.charAt(len - 2) == c)) {
                answer.append(c);
            }
        }

        return answer.toString();
    }
}

File: problem/Make_Fancy_String/makeFancyString.cpp
class Solution
{
public:
    string makeFancyString(string s)
    {
        string str = "";
        int count = 0;
        for (int i = 0; i < s.length(); i++)
        {
            if (i > 1 && count == 2 && s[i] == s[i - 1])
                continue;

            str += s[i];
            if (i > 0 && s[i] == s[i - 1])
                count++;
            else
                count = 1;
        }
        return str;
    }
};